### ISSUE TITLE
targets does not contain software under test

### ISSUE THREAD

**Issue by GalaxyGorilla** [stritzinger/sscg_generator#3]
Under declarations there is the 'targets' array which needs to contain the information about the software under test. This information comes from the SBOM.

**Comment by IciaCarroBarallobre** [comment]
Yes, I think that It was what we did [here](https://github.com/stritzinger/sscg_generator/blob/e480fab7a46b1dc4c9eaf0bcd5c0f223f4d18d86/src/sscg_generator_generate.erl#L126)

`declarations > targets > components` SSCG => SBOM `metadata > component`

**Comment by GalaxyGorilla** [comment]
See email

**Comment by matlaj** [comment]
Any update on this?
I don't think the `declarations > targets > components` section is where we want to identify the software under test in the SSCG. The declarations section is about "The list of declarations which describe the conformance to standards." We might want to add info here later, but for now we should just make sure to fill in `metadata > component`, which is "The component that the BOM describes."

Not sure what to do when the SBOM does not identify the component, though...

**Comment by IciaCarroBarallobre** [comment]
![Image](https://github.com/user-attachments/assets/9ce11910-1550-49f1-b504-63b5707c973d)

The email mentioned

**Comment by matlaj** [comment]
Okay, but the SSCG does not contain in the `metadata > component` field even if the SBOM does have the relevant information.
True, it does fill in `declarations > targets > components`, but I still think that this section is not meant for identifying the subject of the document as a whole, it's just for specifying what certain claims are about.

To illustrate the difference: I imagine we could describe an umbrella Erlang application in the SSCG by describing the "umbrella" itself in `metadata > component`, and then describing all the apps that make up the umbrella in `metadata > component > components`.
Or any software with a microservice architecture could be described the same way.

And then if the security findings are clearly specific to a certain component of the umbrella application, then in the `declarations > targets > components` section the SSCG could identify that subcomponent instead of the umbrella application itself.
